docs/conf.py

```diff
@@ -178,3 +178,55 @@
 
     edit_on_github_source_root = ""
     edit_on_github_doc_root = "docs"
+
+
+# Render rst pages as a jinja template to be able 
+# to generate rst docs from data (i.e. passed to template context) ---------
+def rstjinja(app, docname, source):
+    if app.builder.format != 'html':  # Make sure builder output is HTML
+        return
+    src = source[0]
+    context = app.config.html_context  # context dictionary, passed for rendering
+    context.update(facility_context(docname))
+    rendered = app.builder.templates.render_string(src, context)
+    source[0] = rendered
+
+def facility_context(docname):
+    dirsInPath = docname.split(os.sep)
+    if '_facility_pages' in dirsInPath[:-1]: 
+    # doc file (.rst) lies in directory "_facility_pages"
+        facility_name = dirsInPath[-1]
+        context = {'facility': facility_name}
+    else:
+        context = {}
+    return context
+
+def setup(app):
+    app.connect("source-read", rstjinja)
+
+
+import wsynphot
+import pandas as pd
+import shutil
+
+# Storing data into variables ------------------
+try:
+    df = wsynphot.list_filters()
+except IOError:  # If filter_data.h5 not present
+    wsynphot.download_filter_data()
+    df = wsynphot.list_filters()
+# Obtain list of Obsv. facilities from df
+facility_list = df['Obs. Facility'].unique().tolist()
+
+# Pass data variables to html_context for rendering templates ------
+html_context = {
+    'data': df,
+    'facility_list': facility_list
+}
+
+# Automatically create(or overwrite) doc for filters of each obsv. facility
+# by using facilty_page.rst as template file to populate all facilty pages
+os.makedirs('_facility_pages', exist_ok=True)
+for facility in facility_list:
+    shutil.copyfile('_templates/facility_page.rst',
```

Comment:
is that really necesary?

Reply:
Yes it is absolutely necessary. See, we're building a static documentation of a very large dataset (~4500 filters) so we need to distribute that data across different pages - for which we're putting them hierarchically as per the facilities they belong to. **Since we need ~78 facility HTML pages, thus we have to provide Sphinx ~78 RST pages from which it can build** HTML by rendering them differently using Jinja Template variables. 

This is achieved by a single template file `facility_page.rst` as I explained in 1st comment. Think of `_facilty_pages` dir hence created, equivalent to `_build` dir which are solely generated & controlled by Sphinx while building docs, without requiring any user intervention. The only difference is `_facilty_pages` dir is created as a part of _**pre-build step**_ immediately after command to build docs is given, and `_build` dir when rst is read & rendered. So they are more or less same which make sure that docs can be build, while at the same time they doesn't affect our repo (gitignored).

Reply:
I understand this - I mean can you create them in the right directory

Reply:
What do you think is right directory? 

We can't put them in `_build` or `_template` dir, Sphinx doesn't read files from them. We can't put them in `_static` dir i.e. meant for custom stylesheets and other static files (like images, etc.), not at all for storing .rst files. I've accounted all options then used it and it gives desired results.

Sphinx docs even say that, "Sphinx is highly customizable but that means all you can do with them is not documented". I understand what I've done, is quite new to grasp, but consider it as a customized way to achieve what we want to implement.

I might be missing something, could you please elaborate what wrong do you see in it?

Reply:
it's not very new to grasp 😉 - you somehow create the files earlier and then move them - why not create them directly where they should live.